cerealize/cerealize.h

```diff
@@ -266,8 +381,8 @@ struct AsConstCharPtr {
 
 template <typename OSTREAM, typename T, typename S>
 inline OSTREAM& AppendAsJSON(OSTREAM& os, T&& object, S&& name) {
-  cerealize::CerealStreamType<cerealize::CerealFormat::JSON>::CreateOutputArchive(os)(
-      cereal::make_nvp<typename std::remove_reference<T>::type>(AsConstCharPtr::Run(name), object));
+  cereal::JSONOutputArchive so(os);
+  so(cereal::make_nvp<typename std::remove_reference<T>::type>(AsConstCharPtr::Run(name), object));
```

Comment:
`object` -> `std::forward<remove_ref<...>> object`, which (as always) invites a `typedef` for `remove_reference`.